src/legacy/core_plugins/kibana/public/discover/np_ready/components/field_chooser/field_chooser.js

```diff
@@ -181,11 +183,73 @@ export function createFieldChooserDirective($location, config, $route) {
         $scope.indexPattern.popularizeField(fieldName, 1);
       };
 
+      function getMapsAppUrl() {
+        const services = getServices();
+        const mapsAppVisAlias = services.visualizations.types.getAliases().find(({ name }) => {
+          return name === 'maps';
+        });
+        return mapsAppVisAlias ? mapsAppVisAlias.aliasUrl : null;
+      }
+
+      function isFieldVisualizable(field) {
+        const mapsAppUrl = getMapsAppUrl();
+        if (mapsAppUrl && (field.type === 'geo_point' || field.type === 'geo_shape')) {
+          return true;
+        }
+
+        return field.visualizable;
+      }
+
       function getVisualizeUrl(field) {
         if (!$scope.state) {
           return '';
         }
 
+        const mapsAppUrl = getMapsAppUrl();
+        if (mapsAppUrl && (field.type === 'geo_point' || field.type === 'geo_shape')) {
```

Comment:
Not mandatory, but I think it might make sense to refactor `getMapsAppUrl`, `isFieldVisualizable` and the part of the code responsible for generating the Maps url to an separate file in the `libs` folder of this directory, would improve readability of the code, however could also be done in a later refactoring, since this has to be cleaned up anyway.